packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -331,12 +331,29 @@ export abstract class SfdxCommandletExecutor<T>
       cwd: getRootWorkspacePath()
     }).execute(cancellationToken);
 
-    execution.processExitSubject.subscribe(() => {
-      this.logMetric(execution.command.logName, startTime);
+    let output = '';
+    execution.stdoutSubject.subscribe(realData => {
+      output += realData.toString();
+    });
+
+    execution.processExitSubject.subscribe(async exitCode => {
+      this.logMetric(
+        execution.command.logName,
+        startTime,
+        await this.getTelemetryData(exitCode === 0, response, output)
+      );
     });
     this.attachExecution(execution, cancellationTokenSource, cancellationToken);
   }
 
+  protected getTelemetryData(
```

Comment:
This method can be overridden in child classes to add additional data to a command execution